src/Workspaces/Core/Portable/Serialization/AbstractOptionsSerializationService.cs

```diff
@@ -73,57 +73,57 @@ protected static void WriteCompilationOptionsTo(CompilationOptions options, Obje
             // StrongNameProvider strongNameProvider
         }
 
-        protected static void ReadCompilationOptionsFrom(
+        protected static (
+            OutputKind outputKind,
+            bool reportSuppressedDiagnostics,
+            string moduleName,
+            string mainTypeName,
+            string scriptClassName,
+            OptimizationLevel optimizationLevel,
+            bool checkOverflow,
+            string cryptoKeyContainer,
+            string cryptoKeyFile,
+            ImmutableArray<byte> cryptoPublicKey,
+            bool? delaySign,
+            Platform platform,
+            ReportDiagnostic generalDiagnosticOption,
+            int warningLevel,
+            IEnumerable<KeyValuePair<string, ReportDiagnostic>> specificDiagnosticOptions,
+            bool concurrentBuild,
+            bool deterministic,
+            bool publicSign,
+            MetadataImportOptions metadataImportOptions,
+            XmlReferenceResolver xmlReferenceResolver,
+            SourceReferenceResolver sourceReferenceResolver,
+            MetadataReferenceResolver metadataReferenceResolver,
+            AssemblyIdentityComparer assemblyIdentityComparer,
+            StrongNameProvider strongNameProvider) ReadCompilationOptionsPieces(
```

Comment:
Would defining a struct instead make this a bit less verbose?

Reply:
the moment this is used in two places, that's the first thing i'll do :)